x-pack/test/api_integration/apis/ml/anomaly_detectors/get.ts

```diff
@@ -0,0 +1,222 @@
+        const { body } = await supertest
+          .get(`/api/ml/anomaly_detectors`)
+          .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+          .set(COMMON_HEADERS)
+          .expect(200);
+
+        expect(body.count).to.eql(2);
+        expect(body.jobs.length).to.eql(2);
+        expect(body.jobs[0].job_id).to.eql(`${jobId}_1`);
+        expect(body.jobs[1].job_id).to.eql(`${jobId}_2`);
+      });
+
+      it('should not allow to retrieve jobs for the user without required permissions', async () => {
+        const { body } = await supertest
+          .get(`/api/ml/anomaly_detectors`)
+          .auth(USER.ML_UNAUTHORIZED, ml.securityCommon.getPasswordForUser(USER.ML_UNAUTHORIZED))
+          .set(COMMON_HEADERS)
+          .expect(404);
+
+        expect(body.error).to.eql('Not Found');
+        expect(body.message).to.eql('Not Found');
+      });
+    });
+
+    describe('GetAnomalyDetectorsById', () => {
+      it('should fetch single anomaly detector job by id', async () => {
+        const { body } = await supertest
+          .get(`/api/ml/anomaly_detectors/${jobId}_1`)
+          .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+          .set(COMMON_HEADERS)
+          .expect(200);
+
+        expect(body.count).to.eql(1);
+        expect(body.jobs.length).to.eql(1);
+        expect(body.jobs[0].job_id).to.eql(`${jobId}_1`);
+      });
+
+      it('should fetch anomaly detector jobs based on provided ids', async () => {
+        const { body } = await supertest
+          .get(`/api/ml/anomaly_detectors/${jobId}_1,${jobId}_2`)
+          .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+          .set(COMMON_HEADERS)
+          .expect(200);
+
+        expect(body.count).to.eql(2);
+        expect(body.jobs.length).to.eql(2);
+        expect(body.jobs[0].job_id).to.eql(`${jobId}_1`);
+        expect(body.jobs[1].job_id).to.eql(`${jobId}_2`);
+      });
+
+      it('should not allow to retrieve a job for the user without required permissions', async () => {
+        const { body } = await supertest
+          .get(`/api/ml/anomaly_detectors/${jobId}_1`)
+          .auth(USER.ML_UNAUTHORIZED, ml.securityCommon.getPasswordForUser(USER.ML_UNAUTHORIZED))
+          .set(COMMON_HEADERS)
+          .expect(404);
+
+        expect(body.error).to.eql('Not Found');
+        expect(body.message).to.eql('Not Found');
+      });
+    });
+
+    describe('GetAnomalyDetectorsStats', () => {
+      it('should fetch jobs stats', async () => {
+        const { body } = await supertest
+          .get(`/api/ml/anomaly_detectors/_stats`)
+          .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+          .set(COMMON_HEADERS)
+          .expect(200);
+
+        expect(body.count).to.eql(2);
+        expect(body.jobs.length).to.eql(2);
+        expect(body.jobs[0].job_id).to.eql(`${jobId}_1`);
+        expect(body.jobs[0]).to.keys(
```

Comment:
Nit:I think `to.have.keys` would be nicer to read, although it makes no difference in the validation. Similar for the two `keys` validations below.

Reply:
ok thanks, good point, I'll add it in another API integration tests PR I'm going to create